### ISSUE TITLE
[BUG] Elite Justo 2 is not recognized after first use

### ISSUE THREAD

**Issue by aarongarciah** [cagnulein/qdomyos-zwift#2922]
**Describe the bug**
The first time I opened the app (including when removing and installing the app again), the Elite Justo 2 trainer was recognized. When I open the app again, the screen is empty (no tiles are displayed). See the screenshots below.

I tried setting the _FTMS Bike_ option under the _Bike Options_ section of the settings to "Justo 2", but then only the power is being picked up by the app.

**To Reproduce**
_⚠️ It's hard to reproduce since you need an Elite Justo 2 trainer_

1. Install the app.
2. Do the initial setup using the wizard and configuring Justo 2 as your bike fitness device.
3. Use the trainer and see all values being picked up by the app.
4. Close the app.
5. Open the app and check that no tiles are being shown and no virtual device is created to be used on e.g. Zwift.

**Expected behavior**
The app should show the tiles and create a virtual device to be picked up by e.g. Zwift.

**Screenshots**
First time using the app (or after uninstalling and installing the app again)
<img src="https://github.com/user-attachments/assets/15564efd-31d8-492a-814c-098c7518f0a4" width="375" />

Subsequent usages of the app
<img src="https://github.com/user-attachments/assets/c3a617a8-35f4-40fb-a0c1-a9d964885828" width="375" />

**Desktop (please complete the following information):**

**Smartphone (please complete the following information):**
 - Device: iPhone 12
 - OS: iOS
 - Version 18.2

**Append a debug log**
No debug log is generated in the qdomyoszwift folder in the iOS File app.
  
  
**Additional context**
n/a

**Comment by cagnulein** [comment]
Set the manual device in the advanced settings to.your trainer 

Let me know

**Comment by aarongarciah** [comment]
@cagnulein yes, the "Manual Device" is properly set in the advanced settings, but it doesn't work.

<img src="https://github.com/user-attachments/assets/e89a26e4-57bb-4cf1-b527-beb0d3109a1a" width="375" />

Any idea why no log files are being generated?

**Comment by cagnulein** [comment]
Did you enable the debug log setting in the experimental settings?

**Comment by aarongarciah** [comment]
> Did you enable the debug log setting in the experimental settings? 

I missed that. Here's the debug log: [debug-Thu_Dec_19_19_46_05_2024.log](https://github.com/user-attachments/files/18202021/debug-Thu_Dec_19_19_46_05_2024.log)

**Comment by cagnulein** [comment]
The problem is the ftms bike setting now is disabled, set it and press OK

Let me know!

Roberto Viola
Software engineer and open source enthusiast
http://robertoviola.cloud



Il giorno gio 19 dic 2024 alle 19:49 Aarón García Hervás <
***@***.***> ha scritto:

**Comment by aarongarciah** [comment]
I tried that but the problem it only picks up power and nothing else: no speed, cadence, heart rate, etc.

![IMG_6271](https://github.com/user-attachments/assets/5f48c3d9-9a57-4327-9de5-af3299a7ffcf)

**Comment by aarongarciah** [comment]
Also, any idea why the warning symbol appears on power-related tiles? 
<img width="269" alt="image" src="https://github.com/user-attachments/assets/5760b3fd-b20b-441d-84be-344152315828" />

**Comment by cagnulein** [comment]
The warning symbol is the icon of the wattage in qz, you can disable it in the UI settings of qz

For the metric issues create a debug log https://github.com/cagnulein/qdomyos-zwift/wiki/How-do-i-get-the-debug-log-in-case-something-doesn't-work%3F

**Comment by aarongarciah** [comment]
Looks like it now works 😅 I didn't do anything different. Here's the log: 
[debug-Thu_Dec_19_21_19_19_2024.log](https://github.com/user-attachments/files/18202810/debug-Thu_Dec_19_21_19_19_2024.log)

For the Zwift Click not working, should I open another issue?

**Comment by cagnulein** [comment]
that's great, i guess that maybe you didn't restart the app after changing a setting?

for the click the main issue is that I can't see the click on the log so it means it's not awake when you open qz, isn't it? it must be awake before starting qz

Let me know

**Comment by aarongarciah** [comment]
> that's great, i guess that maybe you didn't restart the app after changing a setting?

I restarted the app and the trainer many times, not sure what happened 😅 

> for the click the main issue is that I can't see the click on the log so it means it's not awake when you open qz, isn't it? it must be awake before starting qz

It's working now. Again, not sure what happened. I swear the Zwift Click was awake before.

Is there a way to set a max number of gears? I see there's no limit or the limit is too high (I tested and it went above 90).

**Comment by cagnulein** [comment]
Glad it worked!
Nope there is no limit with this configuration because so anyone can
increase or decrease the effort freely
In case I can put 2 settings for the limits but qz it's already full of
settings I'm trying to don't add more :D

Il giorno gio 19 dic 2024 alle 22:12 Aarón García Hervás <
***@***.***> ha scritto:

**Comment by aarongarciah** [comment]
@cagnulein I'm confused, how do I know I reached the lowest or higher gear then?

**Comment by cagnulein** [comment]
@aarongarciah the lowest and higher are set internally on the trainer. i mean over and under a certain value of gears in qz the trainer will filter it out. in this way qz is agnostic to the trainer, and you can increase and decrease freely.

**Comment by aarongarciah** [comment]
Ok, thanks!

**Comment by cagnulein** [comment]
Let me know if you have any other feedbacks or suggestions!

Roberto Viola
Software engineer and open source enthusiast
http://robertoviola.cloud



Il giorno ven 20 dic 2024 alle 08:57 Aarón García Hervás <
***@***.***> ha scritto:
